File: detector.js
/**
 * Pair programming starter — implement flagSuspiciousEvents().
 *
 * Each event:
 * {
 *   id: "evt_001",
 *   timestamp: "2025-07-09T14:30:00Z",
 *   actor: { user_id: "alice", ip: "..." },
 *   action: "token.access",
 *   resource: { type: "canary_token", id: "tok_prod_api" },
 *   geo: { country: "US", region: "..." }  // may be null
 * }
 *
 * Return one result per input event:
 * { event_id, suspicious, reasons, severity }
 */

/**
 * @param {object[]} events
 * @returns {object[]}
 */
function flagSuspiciousEvents(events) {
  return events.map((event) => ({
    event_id: event.id,
    suspicious: false,
    reasons: [],
    severity: 0,
  }));
}

module.exports = { flagSuspiciousEvents };

File: detector.py
"""
Pair programming starter — implement flag_suspicious_events().

Each event looks like:
{
  "id": "evt_001",
  "timestamp": "2025-07-09T14:30:00Z",   # ISO-8601 UTC
  "actor": { "user_id": "alice", "ip": "..." },
  "action": "token.access",
  "resource": { "type": "canary_token", "id": "tok_prod_api" },
  "geo": { "country": "US", "region": "..." }   # may be null
}

Return a list of result objects, one per input event:
{
  "event_id": str,
  "suspicious": bool,
  "reasons": list[str],
  "severity": int   # optional until interviewer asks for scoring
}
"""

from __future__ import annotations

from typing import Any


def flag_suspicious_events(events: list[dict[str, Any]]) -> list[dict[str, Any]]:
    """Flag events that match suspicious criteria (define rules with interviewer)."""
    results: list[dict[str, Any]] = []

    for event in events:
        # TODO: implement with interviewer
        results.append(
            {
                "event_id": event["id"],
                "suspicious": False,
                "reasons": [],
                "severity": 0,
            }
        )

    return results
